### PR TITLE
BAH-1093 : Tele-consultation feature merge

### PR DESCRIPTION
**Pull request by praveenadayanand**

### CODE REVIEW

File: api/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -195,6 +195,16 @@
         </sql>
     </changeSet>
 
+    <changeSet id="create-column-teleconsultation-202008241355" author="Vishal">
```
Comment (angshu) [inline api/src/main/resources/liquibase.xml:198]: precondition should be whether the column exists.

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -195,6 +195,16 @@
...
+    <preConditions onFail="MARK_RAN">
+            <tableExists tableName="patient_appointment" />
+    </preConditions>
+    <sql>
```
Comment (angshu) [inline api/src/main/resources/liquibase.xml:202]: use liquibase "addColumn" tag instead of SQL

File: api/src/main/resources/liquibase.xml [hunk 3]
```diff
@@ -342,6 +352,14 @@
         <sqlFile path="patientUpcomingAppointments.sql"/>
     </changeSet>
 
+    <changeSet id="global-property-upcoming-appointments-sql-update" author="Shankar, Mahesh">
```
Comment (angshu) [inline api/src/main/resources/liquibase.xml:355]: add global property in config.xml to add global property. 
example: https://github.com/Bahmni/openmrs-module-appointments/blob/master/omod/src/main/resources/config.xml#L61

File: api/src/main/resources/liquibase.xml [hunk 4]
```diff
@@ -576,4 +594,10 @@
             <column name="description" value="URL pattern to use for published Recurring appointment events. Default is /openmrs/ws/rest/v1/recurring-appointments?uuid={uuid}. If you change default value, please add your custom implementation for the given URL"/>
         </insert>
     </changeSet>
+    <changeSet id="default-202009011116" author="Deedee Lee">
```
Comment (angshu) [inline api/src/main/resources/liquibase.xml:597]: add precondition to check whether email as person attribute type exists

File: api/src/main/resources/patientUpcomingAppointmentsUpdate.sql [hunk 5]
```diff
@@ -0,0 +1,23 @@
+UPDATE global_property
```
Comment (angshu) [inline api/src/main/resources/patientUpcomingAppointmentsUpdate.sql:1]: rename to "patientUpcomingAppointments_v2.sql"

**Review by angshu — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java [hunk 6]
```diff
@@ -10,40 +10,27 @@
...
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
 import java.util.stream.Collectors;
 
 import static java.util.Objects.isNull;
-import static org.openmrs.module.appointments.constants.PrivilegeConstants.MANAGE_APPOINTMENTS;
-import static org.openmrs.module.appointments.constants.PrivilegeConstants.MANAGE_OWN_APPOINTMENTS;
-import static org.openmrs.module.appointments.constants.PrivilegeConstants.RESET_APPOINTMENT_STATUS;
+import static org.openmrs.module.appointments.constants.PrivilegeConstants.*;
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java:29]: replace with single line imports

**Review by angshu — CHANGES_REQUESTED** [review]

File: api/src/main/resources/liquibase.xml (hunk 3 above)
Comment (swatigogia2020) [reply api/src/main/resources/liquibase.xml:355]: This is adding value in the table name global_property and not a global property used in java code.

File: api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java (hunk 6 above)
Comment (praveenadayanand) [reply api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java:29]: @angshu We will incorporate this as part of our release 2.

File: api/src/main/resources/liquibase.xml (hunk 2 above)
Comment (praveenadayanand) [reply api/src/main/resources/liquibase.xml]: Done

File: api/src/main/resources/liquibase.xml (hunk 4 above)
Comment (praveenadayanand) [reply api/src/main/resources/liquibase.xml:601]: Done

**Review by angshu — APPROVED** [review]

**Pull request merged**